### ISSUE TITLE
Selenium 3.0.7 - ChildProcess issue (Compatibility with Cucumber) - OS SPECIFIC

### ISSUE THREAD

**Issue by luke-hill** [SeleniumHQ/selenium#3512]
## Meta -
OS:
Bug found: Windows 7 **yes**, Windows 8 **no**, Windows 10 **no**
Working as intended: Mac / Selenium Grid Extras
Selenium Version:  
`3.0.7` is the buggy version `3.0.5` is stable
Browser:  
Only tested on Chrome

## Expected Behavior -
I can run a suite in RubyMine (Cucumber)
## Actual Behavior -
```
ChildProcess::Error: Access is denied. (5)
./features/support/hooks.rb:29:in `After'
1 scenario (1 failed)
###THEN I GET THIS STACKTRACE###
C:/Ruby22/lib/ruby/gems/2.2.0/gems/childprocess-0.6.1/lib/childprocess/abstract_process.rb:183:in `assert_started': process not started (ChildProcess::Error)
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/childprocess-0.6.1/lib/childprocess/windows/process.rb:39:in `exited?'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/service.rb:124:in `process_exited?'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/service.rb:110:in `stop_process'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/service.rb:83:in `ensure in stop'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/service.rb:83:in `stop'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/service.rb:69:in `block in start'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/platform.rb:138:in `block in exit_hook'
C:/Ruby22/lib/ruby/gems/2.2.0/gems/childprocess-0.6.1/lib/childprocess/abstract_process.rb:183:in `assert_started': process not started (ChildProcess::Error)
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/childprocess-0.6.1/lib/childprocess/windows/process.rb:39:in `exited?'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/service.rb:124:in `process_exited?'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/service.rb:110:in `stop_process'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/service.rb:83:in `ensure in stop'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/service.rb:83:in `stop'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/service.rb:69:in `block in start'
    from C:/Ruby22/lib/ruby/gems/2.2.0/gems/selenium-webdriver-3.0.7/lib/selenium/webdriver/common/platform.rb:138:in `block in exit_hook'
```
## Steps to reproduce -
Self explanatory, try running any feature file with the Webdriver gem set to `3.0.7`

**Comment by titusfortner** [comment]
This is the only change I see: e7d4f5e21e793cc8fd5fb16278d49d07b4f7d660
@luke-hill - you can edit files from external libraries in RubyMine. Go to `rb/lib/selenium/webdriver/chrome/service.rb` and see if removing `@process.leader = true` line gets this to work.

**Comment by titusfortner** [comment]
@p0deje - it's been verified that it's the `@process.leader = true` line that is causing this on Windows 7 & 8. Should we just remove it again?

**Comment by lmtierney** [comment]
It seems to be crapping out on `assign_process_to_job_object` which looking at the clause in `hProcess` [from the microsoft dev center](https://msdn.microsoft.com/en-us/library/windows/desktop/ms681949(v=vs.85).aspx), this would be accurate. 

@luke-hill in IRC you said it was fine on Win8 but in the issue here you said it is happening on Win8 ?

**Comment by p0deje** [comment]
@titusfortner So maybe we can change this to something like this for now

```ruby
@process.leader = true unless Platform.windows?
```

Actually, we should probably look into reporting this bug to childprocess.

**Comment by luke-hill** [comment]
Sorry guys, found out it's not an Issue on Win8 (At least the one I tested). They were using firefox w/o geckodriver so it all crashed (Sep issue, know how to remedy)

**Comment by lmtierney** [comment]
Closed with c861fe7, will be in the next release
